Fix runner non return

kiorky opened the pull request:
If the called function crashes with systemexit, no further event is sent, blocking then the awaitening event waiter function in sync mode.

jacksontj committed:
Make non-async runner executions happen in a thread instead of a daemonized process.

This means we are backwards compatible with builtin print (yay!) and that you can control-c out of the execution (since it will kill the thread too).

jacksontj committed:
Inject globals into *all* runners' global namespace

Fix for #19707
Similar fix to #19712, just doing it once per mod.

jacksontj committed:
Remove unused import

jacksontj committed:
Pylint cleanup

kiorky committed:
Better event exception output

kiorky committed:
also handle systemexit on runner exceptions

cachedout commented:
I encountered this as well. There also seems to be a hiccup in the daemonization call that causes it to hang also. I'll have a PR in shortly.

This looks good, btw, @kiorky. Many thanks!

kiorky commented:
@cachedout if you are going to to major changes, @jacksontj said to me that he is also on is way, for you to know

cachedout commented:
@kiorky Very minor, and not in the runner code itself. I don't think I'll conflict with @jacksontj .

jacksontj committed:
Remove verify_* from all the non-daemon CLI scripts

In the current implementation any call to these would cause the script to traverse *every* salt file on the box. Primarily problematic for salt-run on a master with a large number of minions as part of verify_env touched *each* minion's key

jacksontj committed:
Rework the runner CLI to not use threads

No reason to go through all that work and introduce debugging issues. This way we can still fire events using __jid_event__ but if you just "print" it will do so without trying to monkey patch print

jacksontj committed:
Revert "Fix two problems with runners"

This reverts commit 389fb7e030b3cb340cec6c6d4975d95887819c13.

Conflicts:
	salt/client/mixins.py

jacksontj committed:
Remove unused code

jacksontj commented:
@kiorky I can't seem to cherry pick any of your commits in for some reason, and it no longer merges. I'll just pull in the 2 patches

jacksontj committed:
Nicer error handling for runner execution

jacksontj committed:
Pylint cleanup

kiorky committed:
Revert "Nicer error handling for runner execution"

This reverts commit 5d07de5898ab1dbee67981870e2404e7bf8b2104.

Conflicts:
	salt/client/mixins.py

kiorky commented on salt/client/mixins.py:
```diff
@@ -231,12 +232,10 @@ def low(self, fun, low):
+        except (Exception, SystemExit):
+            trace = traceback.format_exc()
+            data['return'] = 'Exception occurred in {0} {1}:\n{2}'.format(
+                self.client, fun, trace)
```
Exception is printed in the traceback, no need to put it in title !

kiorky committed:
Pylint

kiorky committed:
print event: Do not pop _stamp

kiorky commented:
.. again not applied.
@cachedout i resign ....

kiorky commented:
Well,
